src/shared/watchdog.c

```diff
@@ -84,6 +85,41 @@ static int watchdog_set_timeout(void) {
         return 0;
 }
 
+static int watchdog_get_pretimeout(void) {
+        int sec = 0;
+
+        assert(watchdog_fd > 0);
+
+        if (ioctl(watchdog_fd, WDIOC_GETPRETIMEOUT, &sec) < 0)
+                return log_full_errno(ERRNO_IS_NOT_SUPPORTED(errno) ? LOG_DEBUG : LOG_WARNING, errno, "Failed to get pretimeout value, ignoring: %m");
+
+        watchdog_pretimeout = sec * USEC_PER_SEC;
+
+        return 0;
+}
+
+static int watchdog_set_pretimeout(void) {
+        int sec, r;
+
+        assert(watchdog_fd >= 0);
+        assert(watchdog_pretimeout != USEC_INFINITY);
+
+        sec = saturated_usec_to_sec(watchdog_timeout);
+
+        r = ioctl(watchdog_fd, WDIOC_SETPRETIMEOUT, &sec) < 0;
+        if (r < 0 && ERRNO_IS_NOT_SUPPORTED(r)) {
+                log_info("Watchdog does not support pretimeouts.");
```

Comment:
same thinking as above we should probably set the watchdog_pretimeout variable to 0 here, given the pretimeout is effectively disabled if not supported.